solidpy/Burn.py

```diff
@@ -216,29 +398,26 @@ def raw_simulation_data_export(self):
                     "Regressed Length",
                 ]
             )
-            for (
-                self.time,
-                self.chamber_pressure,
-                self.free_volume,
-                self.regressed_length,
-            ) in zip(
+            for (time, chamber_pressure, free_volume, regressed_length,) in zip(
                 self.solution.t,
                 self.solution.y[0],
                 self.solution.y[1],
                 self.solution.y[2],
             ):
-                self.evaluate_exit_pressure(self.chamber_pressure)
-                self.evaluate_exit_velocity()
-                self.evaluate_thrust(self.chamber_pressure)
+                exit_pressure = self.BurnSimulation.evaluate_exit_pressure(
+                    chamber_pressure
+                )
+                exit_velocity = self.BurnSimulation.evaluate_exit_velocity()
+                thrust = self.BurnSimulation.evaluate_thrust(chamber_pressure)
                 solution_writer.writerow(
                     [
-                        self.time,
-                        self.thrust,
-                        self.chamber_pressure,
-                        self.exit_pressure,
-                        self.exit_velocity,
-                        self.free_volume,
-                        self.regressed_length,
+                        time,
+                        thrust,
+                        chamber_pressure,
+                        exit_pressure,
+                        exit_velocity,
+                        free_volume,
+                        regressed_length,
```

Comment:
Don't you think it's a good idea to store each of these functions as a class attribute?

This way, they can easily be acessed by each post-processing/plotting methods, instead of having to call it from the self.solution array each time.

Reply:
In my latest commits, I changed a bit how the solutions variables work. The class Export reduces many of the unncessary tuples and instance variables that were being utilized, due to the fact that it already has build in Adapter functions so as to provide a clean way to use imported functions (e.g. csv export, mean and max values evaluation etc) in our specific use case.

Of course code refactoring is always a good practice, so I believe it can be improved further in the next versions.